zerver/migrations/0574_backfill_directmessagegroup_group_size.py

```diff
@@ -0,0 +1,72 @@
+# Generated by Django 5.0.6 on 2024-08-16 06:44
+
+from django.db import migrations, transaction
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import Count, Model, OuterRef, Subquery
+
+BATCH_SIZE = 1000
+
+
+def backfill_group_size_field_for_direct_message_groups(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    direct_message_group_model = apps.get_model("zerver", "DirectMessageGroup")
+    recipient_model = apps.get_model("zerver", "Recipient")
+    subscription_model = apps.get_model("zerver", "Subscription")
+
+    RECIPIENT_DIRECT_MESSAGE_GROUP = 3
+
+    direct_message_group_recipient_entries = recipient_model.objects.filter(
+        type=RECIPIENT_DIRECT_MESSAGE_GROUP
+    )
+
+    if not direct_message_group_recipient_entries.exists():
+        return
+
+    recipient_id_lower_bound = direct_message_group_recipient_entries.earliest("id").id
+    recipient_id_upper_bound = direct_message_group_recipient_entries.latest("id").id
+
+    while recipient_id_lower_bound <= recipient_id_upper_bound:
+        do_backfill_group_size_field_for_direct_message_groups(
+            subscription_model,
+            direct_message_group_model,
+            recipient_id_lower_bound,
+            min(recipient_id_lower_bound + BATCH_SIZE, recipient_id_upper_bound),
+        )
+        recipient_id_lower_bound += BATCH_SIZE + 1
+
+
+@transaction.atomic
+def do_backfill_group_size_field_for_direct_message_groups(
+    subscription_model: type[Model],
+    direct_message_group_model: type[Model],
+    recipient_id_lower_bound: int,
+    recipient_id_upper_bound: int,
+) -> None:
+    direct_message_group_sub_size = (
+        subscription_model._default_manager.filter(recipient=OuterRef("recipient"))
```

Comment:
Use `.objects.filter`, not `._default_manager.`.

Reply:
`mypy` doesn't seem to be happy if I use `.objects` for a `type[Model]` of the model passed into a function.

Reply:
Yeah, I guess that makes sense, this works as-is.